Refactor the branding documentation (#73)
owncloud/docs-client-branding: modules/ROOT/pages/ios_app/publishing.adoc

= Publishing Your Branded iOS App
:toc: right

> :description: This documentat guides you thru publishing the branded iOS app. 

Reviewer: ```suggestion
:description: This document guides you through publishing the branded iOS app. 
```

:screenshot-submission-process-url: https://developer.apple.com/news/?id=08082016a
:screenshot-submission-process-video-url: https://developer.apple.com/videos/play/wwdc2016/305/?time=1700
:sized-frames-url: https://itunes.apple.com/app/owncloud/id543672169

== Introduction

{description}

== Publish for General Distribution on iTunes

> At last, after following all the previous steps and passing beta testing, your branded iOS app is ready to publish for general distribution on iTunes. You need a Mac computer with Xcode installed (Xcode is a free download), and you need the xref:ios_app/preparation/provisioning_profiles.adoc[eight provisioning profiles] (4 Ad Hoc provisioning profiles and 4 Apple Store profiles) and P12 certificate file that you created copied to the same computer that you are using to upload your app to iTunes. You will also need a number of screenshots of your app in specific sizes and resolutions, which are detailed in your iTunes Connect setup screen.

Reviewer 2: ```suggestion
At last, after following all the previous steps and passing beta testing, your branded iOS app is ready to be published for general distribution on iTunes. You need a Mac computer with Xcode installed (Xcode is a free download), and you need the xref:ios_app/preparation/provisioning_profiles.adoc[eight provisioning profiles] (4 ad hoc provisioning profiles and 4 app store profiles) and the P12 certificate file that you created and copied to the same computer that you are using to upload your app to iTunes. You will also need a number of screenshots of your app in specific sizes and resolutions, which are detailed in your iTunes Connect setup screen.
```

TIP: Apple must review and approve your app, and the approval process can take several days to several weeks.

Download the `xcarchive.zip` file from your account at {customer-owncloud-url}[customer.owncloud.com]. macOS  will automatically unpack it and change the name to something like `ownCloud iOS Client 02-07-23 10.30.xcarchive`. Double-click on this file to automatically install it into Xcode. Go to Xcode and you will see it in the Archives listing under menu:Window[Organizer].

image::ios_app/publishing/ios-publish-2.png[image, width=450]

> Next, go back to the https://developer.apple.com/membercenter/index.action[Apple Developer Member Center] to log into iTunes Connect to set up your app storefront.

Reviewer: ```suggestion
Next, go back to the https://developer.apple.com/membercenter/index.action[Apple Developer Member Center] to log in to iTunes Connect to set up your app storefront.
```

image::ios_app/publishing/ios-publish-3.png[image, width=450]

> After logging in click the blue btn:[My Apps] button. This takes you to the main screen for managing your apps on iTunes.

Reviewer 2: ```suggestion
After logging in, click the blue btn:[My Apps] button. This takes you to the main screen for managing your apps on iTunes.
```

image::ios_app/publishing/ios-publish-9.png[image]

Apple simplified {screenshot-submission-process-url}[the screenshot submission process]. Please {screenshot-submission-process-video-url}[check this Video (in Safari)] for details.

For the ownCloud client, we also don't use real screenshots, we use {sized-frames-url}[frames in different sizes instead]. You can find templates to generate those assets. Here are examples for the Sketch app:

* https://github.com/LaunchKit/SketchToAppStore
* https://github.com/MengTo/AppStoreSketch

Then you must enter:

* Your app name
* A description
* Some keywords for iTunes searches; and
* Some optional URLs

image::ios_app/publishing/ios-publish-10.png[image, width=450]

The next section is for Apple Watch. If you don't support Apple Watch you can skip this.

The menu:General App Information[] section requires a:

* 1024 x 1024 logo
* Version
* Rating
* Category
* License
* Copyright, and
* Contact information

image::ios_app/publishing/ios-publish-11.png[image, width=450]

In the menu:Build[] section, click the plus button and select your app.

image::ios_app/publishing/ios-publish-14.png[image, width=450]

> The menu:App Review Information[] requires contact information, and some information about your app to guide reviewers. Remember, everyone on iTunes can review your app, so it's in your best interest to be helpful. You may optionally provide a login for a demo account.

Reviewer 2: ```suggestion
The menu:App Review Information[] requires contact information and some information about your app to guide reviewers. Remember, everyone on iTunes can review your app, so it's in your best interest to be helpful. You may optionally provide a login for a demo account.
```

image::ios_app/publishing/ios-publish-12.png[image, width=450]

The menu:Version Release[] section allows you to choose between automatic release, which means your app will be published upon approval, or manual release, where you must release your app after it is approved.

== Pricing and Territories

> Next, you must go to the menu:Pricing[] page to set your price, and to select the territories you want your app to be available in.

Reviewer 2: ```suggestion
Next, you must go to the menu:Pricing[] page to set your price and to select the territories you want your app to be available in.
```

image::ios_app/publishing/ios-publish-13.png[image, width=450]

== Submit For Review

> When you have filled in all the required forms and provided the required screenshots, click btn:[Save] and then *Submit for Review*. If anything needs to be corrected you will see messages telling you exactly what must be fixed.

Reviewer 2: ```suggestion
When you have filled in all the required forms and provided the required screenshots, click btn:[Save] and then *Submit for Review*. If anything needs to be corrected, you will see messages telling you exactly what must be fixed.
```

The next screen is legalese; click the appropriate Yes or No boxes, and then click the btn:[Submit] button.

> You are now finished. No really, you are. When you return to your menu:My Apps[] page you'll see that the status of your app has changed to *Waiting for review*. In a few days, or perhaps many days, your app will either be approved or rejected. If it is rejected Apple will tell you what you need to do to get it approved.

Reviewer 2: ```suggestion
You are now finished. No really, you are. When you return to your menu:My Apps[] page you'll see that the status of your app has changed to *Waiting for review*. In a few days, or perhaps many days, your app will either be approved or rejected. If it is rejected, Apple will tell you what you need to do to get it approved.
```
